fdbclient/ClientLogEvents.h

```diff
@@ -22,15 +22,68 @@
 #ifndef FDBCLIENT_CLIENTLOGEVENTS_H
 #define FDBCLIENT_CLIENTLOGEVENTS_H
 
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/JSONDoc.h"
+
+#include <algorithm>
+
+struct RequestStats {
+	RequestStats() : nextReadId(0) {}
+
+	struct ReadStats {
+		ReadStats(int requestId, int bytesFetched, int keysFetched, Key beginKey, Key endKey, NetworkAddress storageContacted) :
+			requestId(requestId), bytesFetched(bytesFetched), keysFetched(keysFetched), beginKey(beginKey), endKey(endKey), storageContacted(storageContacted) {}
+
+		int requestId;
+		int bytesFetched;
+		int keysFetched;
+		Key beginKey;
+		Key endKey;
+		NetworkAddress storageContacted;
+	};
+
+	std::vector<ReadStats> reads;
+	std::vector<NetworkAddress> proxies;
+
+	int getNextReadId() { return nextReadId++; };
+
+	json_spirit::Object getJson() const {
+		json_spirit::Array proxiesContacted;
+		for (const auto &p : proxies) {
+			proxiesContacted.push_back(json_spirit::Value(p.toString()));
+		}
+		json_spirit::Array readStatsList;
+		for (const auto &r : reads) {
+			json_spirit::Object readStats;
+			readStats.push_back(json_spirit::Pair("requestId", json_spirit::Value(static_cast<uint64_t>(r.requestId))));
+			readStats.push_back(json_spirit::Pair("beginKey", json_spirit::Value(r.beginKey.toString())));
+			readStats.push_back(json_spirit::Pair("endKey", json_spirit::Value(r.endKey.toString())));
+			readStats.push_back(json_spirit::Pair("storageContacted", json_spirit::Value(r.storageContacted.toString())));
+			readStats.push_back(json_spirit::Pair("keysFetched", json_spirit::Value(static_cast<uint64_t>(r.keysFetched))));
+			readStats.push_back(json_spirit::Pair("bytesFetched", json_spirit::Value(static_cast<uint64_t>(r.bytesFetched))));
+			readStatsList.push_back(readStats);
+		}
+		json_spirit::Object result;
+		result.push_back(json_spirit::Pair("readStatistics", json_spirit::Value(readStatsList)));
+		result.push_back(json_spirit::Pair("proxiesContacted", json_spirit::Value(proxiesContacted)));
+		return std::move(result);
+	}
+private:
+	int nextReadId;
+};
+
 namespace FdbClientLogEvents {
 	typedef int EventType;
 	enum {	GET_VERSION_LATENCY	= 0,
-			GET_LATENCY			= 1,
-			GET_RANGE_LATENCY	= 2,
-			COMMIT_LATENCY		= 3,
-			ERROR_GET			= 4,
-			ERROR_GET_RANGE		= 5,
-			ERROR_COMMIT		= 6,
+			GET_VALUE			= 1,
+			GET_KEY			= 2,
+			GET_RANGE_FINISHED	= 3,
+			GET_RANGE			= 4,
```

Comment:
Is the idea here that `GET_RANGE_FINISHED` captures the entire range request, while `GET_RANGE` records info for individual range queries to the storage server? I think it may be more clear to a client of the database if these were named `GET_RANGE` for the whole request and something like `GET_SUBRANGE` for the internal partial range query. In particular, this lines up the name `GET_RANGE` with the `getRange` call that the client is making.